beacon-chain/rpc/apimiddleware/custom_hooks.go

```diff
@@ -17,6 +17,29 @@ import (
 	"github.com/prysmaticlabs/prysm/v3/time/slots"
 )
 
+// https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/submitPoolBLSToExecutionChange
+// expects posting a top-level array. We make it more proto-friendly by wrapping it in a struct.
+func wrapBLSChangesArray(
+	endpoint *apimiddleware.Endpoint,
+	_ http.ResponseWriter,
+	req *http.Request,
+) (apimiddleware.RunDefault, apimiddleware.ErrorJson) {
+	if _, ok := endpoint.PostRequest.(*SubmitBLSToExecutionChangesRequest); !ok {
+		return true, nil
```

Comment:
Should this return true, nil if !ok?

Reply:
Every other wrapper does this so I'll follow this pattern so I didn't want to change it, pinging @rkapka that may have had a reason.  This can never happen anyway in runtime.